docs/CodeDocumentationGuidelines.md

```diff
@@ -0,0 +1,28 @@
+# Code Documentation Guidelines
+
+In this document you will find the guidelines for adding XML comments to our codebase. By adding consistent and well-formatted comments to our code we benefit on all fronts: the online API docs are a useful reference for people looking up our APIs, the IntelliSense inside of Visual Studio will help developers understand our product better and contributors and maintainers of this repository will have all the documentation at hand.
+
+## Guidelines
+
+For what kind of comments to our code we mainly follow the [recommended XML tags documentation](https://learn.microsoft.com/dotnet/csharp/language-reference/xmldoc/recommended-tags) by Microsoft. This is also what is best supported by Visual Studio. As a rule of thumb: complete your code first and just start typing a tripe slash (`///`) on top of your code. That will suggest all the attributes that we want to see. Which attributes will show up is infered from your code.
+
+If you're unsure about how to document a certain element, have a look the [.NET API docs wiki](https://github.com/dotnet/dotnet-api-docs/wiki) which has a very extensive description on what kind of comment to add on which element in the code. We would highly recommend going through that and apply the same style of comments everywhere.
+
+These are the tags that we would like to see when applicable: `<summary>`, `<remarks>`, `<returns>`, `<param>`, `<exception>`, `<inheritdoc>`, `<see>`, `<c>`.
+
+* All public members should have at the very least a `<summary>`
+* Keep the descriptions short and consise
+  * 1-2 lines typically, no screenshots or long code-blocks, that needs to go to conceptual docs
+* Make sure the spelling is correct
+* Add the descriptions at the highest base class/interface level. On an implementing type add `<inheritdoc>` on each member
+  * If the implemented member differs to much you can choose to override the comments, typically shouldn't be necessary
+  * Do **not** add `<inheritdoc>` to overridden members, this will potentially cause issues in the online API docs system and doesn't add any value. The documentation is inherited implicitly.
+* Where applicable make references to other types and parameters with the appropriate tags (`<see cref="YourType">` and `<paramref.........` respectively). This will make links in IntelliSense and online API docs
```

Comment:
```suggestion
* Where applicable make references to other types and parameters with the appropriate tags (`<see cref="YourType">` and `<paramref.........` respectively). This will create links in IntelliSense and online API docs
```